acc.php

```diff
@@ -2035,6 +2035,43 @@
 		}
 	});
 }
+elseif ($action == "listall") {
+    global $availableRequestStates, $enableEmailConfirm;
+
+    $database = gGetDb();
+
+    if ($enableEmailConfirm == 1) {
+        $query = "SELECT * FROM request WHERE status = :type AND emailconfirm = 'Confirmed';";
+    }
+    else {
+        $query = "SELECT * FROM request WHERE status = :type;";
+    }
+
+    $statement = $database->prepare($query);
+
+    if (isset($_GET['status']) && isset($availableRequestStates[$_GET['status']])) {
```

Comment:
Maybe it makes sense to move this conditional up to just after the `global $availableRequestStates`?  That way we don't have to spend time preparing the database statement unless we have everything we need.